### ISSUE TITLE
Edit profile: Inconsistent error messages (styling); messages disappear after a few seconds

### ISSUE THREAD

**Issue by qanitah** [mmiron-HC/WST-RSAMS#45]
URL: https://hcdev.openplus.ca/en/user/203/edit
User Agent: Chrome, Firefox
Language: both

Issue: As mentioned in issue #35, error messages are not following WET format. Along with with issue, there are different formats used and error messages are inconsistent. The only time the list of errors shows up on top is when the error is with Password field. All errors should be listed on top. The error message when there is an error in required field, format is incorrect, mismatched passwords etc.

Expected result: Content should be consistent, should be displayed clearly and remain on screen for the user. The error message should be between the label and the input field. When the error appears, it should be part of the label. 

![image](https://user-images.githubusercontent.com/14280062/128538533-d0f3e636-c162-469b-a751-fd45f52f4d1d.png)

![image](https://user-images.githubusercontent.com/14280062/128538637-a1289cfe-ea28-4588-99ec-789180f58829.png)

![image](https://user-images.githubusercontent.com/14280062/128538752-67e56eaf-6665-4300-9130-cd39378acdc4.png)

**Comment by davidgotlieb** [comment]
﻿These aren't really error messages. Tooltips were there to guide users filling out forms. Tooltips have been removed across the site.

**Comment by qanitah** [comment]
It still appears while it should have an error message. Also, the page scrolls up when user clicks 'save' but only the message appears and field is not visible. I thought tooltips were the message in black block but this is different.
![image](https://user-images.githubusercontent.com/14280062/136228880-b9be5c81-8bcb-42ba-8ef7-911491a4b802.png)

Same for username field. It should be a proper error message. It also doesn't scroll up properly to have the field in view.
![image](https://user-images.githubusercontent.com/14280062/136229041-5f392a46-ddb9-4d90-b3e4-fe079576fdb0.png)

**Comment by echouin** [comment]
@qanitah From OP: I did some investigating into this. Those tips are coming from the browser. They have nothing to do with the site. I ﻿would recommend closing this card.﻿

**Comment by mmiron-HC** [comment]
@echouin We should not rely on the browser validation for several reasons: 
1. browser language will be an issue. if my browser language is English and I'm working in the French interface and trigger errors, the errors will be displayed in English. This is a failure.
2. The errors are not persistent and disappear quickly. This will cause issues for users who have cognitive difficulties as they will not remember what/why the error was triggered, 
3. Errors should also all be triggered at once. 
This is why we suggest following WET where the error messages are identified clearly at the top of the page. The reason for this is that normally the screen reader users will be brought back up to the top of the page therefore there should be a message at the top.
we also suggest that the error messages appear between the field label and the input field, again this will help with screen reader users but also with low vision users who are using a magnifier, they may miss that an error has been triggered if it is below.

**Comment by echouin** [comment]
From OP:  ﻿I'm recommending this for closure. It only affects the email field for logged in users. If it's a significant issue, we can move it to the parking lot and deal with it in a future sprint.﻿

**Comment by davidgotlieb** [comment]
Amount of work to fix this would be massive, as we would have to completely rework how Drupal does error handling. Exploring if this is something we can report to WxT or Drupal core to replace browser error handling with templated error handling from Drupal.

**Comment by echouin** [comment]
From OP: Resolved. On Dev to test. Should be on latest release and live on Public site on 12 Mar 2022.

**Comment by hydegeoff** [comment]
I believe issue can be closed.  There no more pop-up error messages appearing and the screen reader announces the errors when navigating to the fields
